src/test/java/com/wavefront/sdk/common/clients/WavefrontClientTest.java

```diff
@@ -64,6 +79,112 @@ private String createString(int size) {
     return new String(new char[size]).replace("\0", "a");
   }
 
+  @Nested
+  class WireMockTests {
+    WireMockServer mockBackend;
+
+    @BeforeEach
+    void setup() {
+      mockBackend = new WireMockServer(wireMockConfig().dynamicPort());
+      mockBackend.stubFor(WireMock.post(urlPathMatching("/report")).willReturn(WireMock.ok()));
+      mockBackend.start();
+    }
+
+    @AfterEach
+    void teardown() {
+      mockBackend.stop();
+    }
+
+    @Test
+    void sendMetric() {
+      WavefrontClient wfClient = new WavefrontClient.Builder(mockBackend.baseUrl())
+          .includeSdkMetrics(false)
+          .build();
+      long timestamp = System.currentTimeMillis();
+
+      assertDoesNotThrow(() -> {
+        wfClient.sendMetric("a-name", 1.0, timestamp, "a-source", new HashMap<>());
+        wfClient.flush();
+      });
+
+      String expectedBody = "\"a-name\" 1.0 " + timestamp + " source=\"a-source\"\n";
+      mockBackend.verify(1, postRequestedFor(urlEqualTo("/report?f=wavefront"))
+          .withRequestBody(matching(expectedBody))
+          .withHeader("Content-Type", WireMock.equalTo("application/octet-stream")));
+    }
+
+    @Test
+    void sendSpan() {
+      WavefrontClient wfClient = new WavefrontClient.Builder(mockBackend.baseUrl())
+          .includeSdkMetrics(false)
+          .build();
+      long timestamp = System.currentTimeMillis();
+      UUID traceId = UUID.fromString("01010101-0101-0101-0101-010101010101");
+      UUID spanId = UUID.fromString("00000000-0000-0000-0000-000000000001");
+
+      assertDoesNotThrow(() -> {
+        wfClient.sendSpan("a-name", timestamp, 1138, "a-source", traceId, spanId,
+            null, null, null, null);
+        wfClient.flush();
+      });
+
+      String expectedBody = "\"a-name\" source=\"a-source\" traceId=" + traceId + " spanId=" +
+          spanId + " " + timestamp + " 1138\n";
+      mockBackend.verify(1, postRequestedFor(urlEqualTo("/report?f=trace"))
+          .withRequestBody(matching(expectedBody))
+          .withHeader("Content-Type", WireMock.equalTo("application/octet-stream")));
+    }
+
+    @Test
+    void sendDistribution() {
+      WavefrontClient wfClient = new WavefrontClient.Builder(mockBackend.baseUrl())
+          .includeSdkMetrics(false)
+          .build();
+      long timestamp = System.currentTimeMillis();
+
+      assertDoesNotThrow(() -> {
+        wfClient.sendDistribution("a-name",
+            Collections.singletonList(new Pair<>(1.1, 2)),
```

Comment:
It might be worth expanding this to multiple buckets at some point, since the histogram format is a little arcane